enhancements/console/customize-perspectives.md

```diff
@@ -0,0 +1,292 @@
+---
+title: customize-perspectives
+authors:
+  - "@lokanandaprabhu"
+reviewers:
+  - "@jerolimov"
+  - "@spadgett"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+api-approvers:
+  - "@JoelSpeed"
+creation-date: 2022-07-26
+last-updated: 2022-08-25
+tracking-link: 
+  - https://issues.redhat.com/browse/ODC-6695
+status: implementable
+---
+
+# Customize Perspectives 
+
+## Summary
+
+OpenShift Container Platform (OCP) web console provides two perspectives: Administrator and Developer. 
+The Developer perspective provides workflows specific to developer use cases 
+like create, deploy and monitor applications, while Administrator perspective 
+is responsible for managing the cluster resources, users, and projects.
+
+This document provides details about how cluster administrator 
+can hide user perspectives. 
+
+## Motivation
+
+Cluster administrator wants to hide the "Administrator" perspective 
+from non-privileged and to hide the "Developer" 
+perspective for all users.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, I want to hide "Administrator" perspective 
+from all non-privileged users so that non-privileged users
+cannot access "Administrator" perspective for managing the cluster resources, 
+users, and projects. Non-privileged users are users that cannot get
+all `namespaces` (based on RBAC check)
+
+#### Story 2
+
+As a cluster admin, I want to hide "Developer" perspective
+for all users.
+
+#### Story 3
+
+As a cluster admin, I want to hide any perspective contributed by a plugin.
+
+Additional perspectives can be dynamically added with the 
+[Perspective extension](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/extensions/perspectives.ts).
+
+### Goals
+
+- Allow cluster admins to hide the user perspectives.
+- Provide a way that cluster admins can do this based on resource permissions (RBAC).
+
+### Non-Goals
+
+- Allow cluster admins to customize the navigation menu along with perspectives for users or groups.
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console` CRD. 
+We can add 
+`spec.customization.perspectives` which can have list of customization details.
+
+### Workflow Description
+
+To disable perspective for users with limited permissions we need a list of options with a perspective id and the required permissions.
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── requiresAccessReview
+|               ├── kind
+|               ├── verb
+└── ...
+```
+
+This might be extendable in the future if we want replace (hide one, show another) a perspective based on a resource permissions:
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── requiresAccessReview
+|           |   ├── kind
+|           |   ├── verb
+|           ├── missingAccessReview
+|               ├── kind
+|               ├── verb
+└── ...
+```
```

Comment:
The other customizations that we have today have the structure
```
customization:
 developerCatalog:
  categories:
  - id:....

and

customization:
  projectAccess:
   availableClusterRoles:
   - ....

and

customization:
 addPage:
  disabledActions:
  - ....
```
So, shouldn't the structure here be something like
```
customization:
  perspectives: 
   <xyz>:
   - id:.....

or

customization:
 <xyz>:
    perspectives: 
    - id:.....
```
    
@spadgett @jerolimov